libraries/AP_Beacon/AP_Beacon_Nooploop.cpp

```diff
@@ -0,0 +1,250 @@
+    _last_update_ms = AP_HAL::millis();
+
+    // estimated precision for x,y position in meters
+    const float precision_x = _msgbuf[NOOPLOOP_NODE_FRAME2_PRECISION_X] * 0.01;
+    const float precision_y = _msgbuf[NOOPLOOP_NODE_FRAME2_PRECISION_Y] * 0.01;
+    const float pos_err = sqrtf(sq(precision_x)+sq(precision_y));
+
+    // x,y,z position in m*1000 in ENU frame
+    const int32_t pos_x = ((int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSX+2] << 24 | (int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSX+1] << 16 | (int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSX] << 8) >> 8;
+    const int32_t pos_y = ((int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSY+2] << 24 | (int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSY+1] << 16 | (int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSY] << 8) >> 8;
+    const int32_t pos_z = ((int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSZ+2] << 24 | (int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSZ+1] << 16 | (int32_t)_msgbuf[NOOPLOOP_NODE_FRAME2_POSZ] << 8) >> 8;
+
+    // position scaled to meters and changed to NED
+    const Vector3f pos_m {pos_y * 0.001f, pos_x * 0.001f, -pos_z * 0.001f};
+
+    set_vehicle_position(pos_m, pos_err);
+
+    const uint8_t valid_nodes = _msgbuf[NOOPLOOP_NODE_FRAME2_VALID_NODES];
+    for (uint8_t i = 0; i < valid_nodes; i++) {
+        uint16_t offset = NOOPLOOP_NODE_FRAME2_NODE_BLOCK + i * 13;
+        uint8_t id = _msgbuf[offset+1];
+        //nooploop id starts from 0, increments clockwise, 0 -> 1 define Y axis. swap 2 and 3 to match pozyx order
+        if (id == 2) {
+            id = 3;
+        } else if (id == 3) {
+            id = 2;
+        }
+        const int32_t dist = ((int32_t)_msgbuf[offset+2+2] << 24 | (int32_t)_msgbuf[offset+2+1] << 16 | (int32_t)_msgbuf[offset+2] << 8) >> 8;
+        set_beacon_distance(id, dist * 0.001f);
+    }
+}
+
+void AP_Beacon_Nooploop::parse_setting_frame0()
+{
+    for (uint8_t i = 0; i < 4; i++) {
+        uint16_t offset = NOOPLOOP_SETTING_FRAME0_A0 + i * 9;
+
+        // x,y,z position in m*1000 in ENU frame
+        const int32_t pos_x = ((int32_t)_msgbuf[offset+2] << 24 | (int32_t)_msgbuf[offset+1] << 16 | (int32_t)_msgbuf[offset] << 8) >> 8;
+        if (pos_x == NOOPLOOP_INVALID_VAL) { //anchor position not available
+            return;
+        }
+        offset+=3;
+        const int32_t pos_y = ((int32_t)_msgbuf[offset+2] << 24 | (int32_t)_msgbuf[offset+1] << 16 | (int32_t)_msgbuf[offset] << 8) >> 8;
+        if (pos_y == NOOPLOOP_INVALID_VAL) {
+            return;
+        }
+        offset+=3;
+        const int32_t pos_z = ((int32_t)_msgbuf[offset+2] << 24 | (int32_t)_msgbuf[offset+1] << 16 | (int32_t)_msgbuf[offset] << 8) >> 8;
+        if (pos_z == NOOPLOOP_INVALID_VAL) {
+            return;
+        }
+
+        // position scaled to meters and changed to NED
+        const Vector3f pos_m {pos_y * 0.001f, pos_x * 0.001f, -pos_z * 0.001f};
+        
+        //nooploop id starts from 0, increments clockwise, 0 -> 1 define Y axis. swap 2 and 3 to match pozyx order
```

Comment:
By the way, the "N" order used for Pozyx is a bit arbitrary. The EKF doesn't care about the order so keeping it as clockwise probably would have made more sense.  I'm happy either way though, this isn't a blocker.